snapcraft/internal/errors.py

```diff
@@ -415,15 +415,43 @@ def __init__(self, image_info):
 
 
 class PatcherError(SnapcraftError):
+    pass
+
+
+class PatcherGenericError(PatcherError):
 
     fmt = (
         '{elf_file!r} cannot be patched to function properly as a classic '
-        'snap: {message}')
+        'snap: {message}'
+    )
 
-    def __init__(self, *, elf_file, message):
+    def __init__(self, *, elf_file, process_exception):
+        message = '{} failed with exit code {}'.format(
+                ' '.join(process_exception.cmd),
+                process_exception.returncode)
         super().__init__(elf_file=elf_file, message=message)
 
 
+class PatcherNewerPatchelfError(PatcherError):
+
+    fmt = (
+        '{elf_file!r} cannot be patched to function properly as a classic '
```

Comment:
You probably want "in" here as well.

This message overall doesn't really explicitly say why this failed, although it's alluded to at the end-- it goes straight from what happened to how to fix it without saying why what happened is a problem. Could we make it say something about "patchelf version {patchelf_version} is too old, Snapcraft requires at least version {good_patchelf_version}" ?